src/test/java/com/suse/saltstack/netapi/client/SaltStackClientTest.java

```diff
@@ -147,6 +155,46 @@ public void testRunRequestAsync() throws Exception {
     }
 
     @Test
+    public void testRunRequestWithSocketTimeout() throws Exception {
+        exception.expect(SaltStackException.class);
+        exception.expectMessage(containsString("Read timed out"));
+
+        // create a local SaltStackClient with a fast timeout configuration 
+        // to do not lock tests more thant 2s
+        URI uri = URI.create("http://localhost:" + Integer.toString(MOCK_HTTP_PORT));
+        SaltStackClient clientWithFastTimeout = new SaltStackClient(uri);
+        clientWithFastTimeout.getConfig().put(SOCKET_TIMEOUT, 1000);
```

Comment:
Why is this a bug @isbm? He suggests to use 1 second here (and below) to not block the tests for 10 seconds. Makes sense to me actually.